lib/ex_doc/markdown.ex

```diff
@@ -52,8 +52,14 @@ defmodule ExDoc.Markdown do
   @doc """
   Changes the markdown processor globally.
   """
-  def put_markdown_processor(processor, options \\ []) do
+  def put_markdown_processor(processor) do
     Application.put_env(:ex_doc, @markdown_processor_key, processor)
```

Comment:
Instead of having two keys and functions, let’s store it as a tuple {processor, options}. WDYT?

Reply:
The `:markdown_processor_options` can be supplied without a `:markdown_processor` key (relying on the default implementation, `ExDoc.Markdown.Earmark` in this case).

What value should `{processor, options}` have if there is no `processor` set?

The current behavior is to discover the `processor` when no value is set:

`get_markdown_processor/0`:

https://github.com/linkdd/ex_doc/blob/patch-1/lib/ex_doc/markdown.ex#L40-L50

`find_markdown_processor/0`:

https://github.com/linkdd/ex_doc/blob/patch-1/lib/ex_doc/markdown.ex#L66-L70

This way, I'm sure I'm not breaking anything 😄

Reply:
The options are specific to a processor anyway, so passing them with the processor is the way to go.

Reply:
To be clear, we could make
The markdown_processor to be a tuple or an atom. If a tuple, the second element is options. :)

Reply:
I opted to always set a tuple but with `processor` set to nil, and if it is nil, we try to resolve it.

Reply:
And we remove markdown_processor_options since it isn’t used anyway. :)

Reply:
So `markdown_processor: {nil, options}` in the Mix configuration instead of 2 options ?

Reply:
Yes, but I don’t want to allow nil. If we change the default, then your options may not work with the new default, so better to be explicit.

Reply:
So, the configuration would be one of:

 - `markdown_processor: ExDoc.Markdown.Earmark`
 - `markdown_processor: [module: ExDoc.Markdown.Earmark, options: ...]`
 - `markdown_processor: [options: ...]`

Reply:
It should be module() or {module(), keyword()}.

Reply:
Ok done 🙂